ui-react/packages/atlasmap-standalone/src/useSingleInputDialog.tsx

```diff
@@ -0,0 +1,63 @@
+import { Modal, Button, TextInput } from '@patternfly/react-core';
+import React, { ReactChild, ReactPortal, useCallback, useState } from 'react';
+import { createPortal } from 'react-dom';
+
+export type Callback = (closeDialog: () => void) => void;
+
+export interface IUseSingleInputDialogArgs {
+  title: string;
+  content: ReactChild;
+  placeholder: string;
+  state: {
+    inputValue: ''
+  };
+  onConfirm: Callback;
+  onCancel: Callback;
+}
+
+export function useSingleInputDialog({
+  title,
+  content,
+  placeholder,
+  state,
+  onConfirm,
+  onCancel,
+}: IUseSingleInputDialogArgs): [ReactPortal, () => void] {
+  const [isOpen, setIsOpen] = useState(false);
+  const openModal = () => setIsOpen(true);
+  const closeModal = () => setIsOpen(false);
+  const handleTextInputChange = (value: string) => {
+    console.log('>>>>>>>>>>>>>>>>>>>>>> handleTextInputChange - value is ' + value);
+    // setState({ value });
+  };
+
+  const handleConfirm = useCallback(() => onConfirm(closeModal), [onConfirm]);
+  const handleCancel = useCallback(() => onCancel(closeModal), [onCancel]);
+  const { inputValue } = state;
```

Comment:
State in React isn't like in Angular, you have to explicitly manage it. So you want to tell React that you need some state:

`const [value, setValue] = useState('');`

and then you want your `TextInput` to use `value` as the value and to trigger `setValue` when handling changes.

Eventually, you'll want to forward `value` to the `onConfirm` callback, so that the caller can deal with it.

Make sure to go through this https://reactjs.org/docs/forms.html